test/upgrade.js

```diff
@@ -0,0 +1,84 @@
+const { expect, assert } = require("chai");
+const { ethers, upgrades } = require("hardhat");
+
+describe("Contract Upgrades", function () {
+  let Token;
+  let arrowToken;
+  let TokenV2;
+  let arrowTokenV2;
+  let owner;
+  let signer1;
+
+  beforeEach(async function () {
+    Token = await ethers.getContractFactory("ArrowToken");
+    [owner, signer1] = await ethers.getSigners();
+    let name = "Arrow Token";
+    let symbol = "ARROW";
+    arrowToken = await upgrades.deployProxy(
+      Token,
+      [ethers.BigNumber.from(1_000_000), name, symbol],
+      { kind: "uups" }
+    );
+    await arrowToken.deployed();
+    TokenV2 = await ethers.getContractFactory("ArrowTokenV2");
+    arrowTokenV2 = await upgrades.upgradeProxy(arrowToken, TokenV2, {
+      call: { fn: "initializeV2", args: [52] },
+    });
```

Comment:
It's weird to me that we're doing the upgrade in the `beforeEach` call, when the upgrade is the thing we're specifically testing here.

Could we move this into the test flow and make sure it succeeds with an `expect`?

Reply:
What are we expecting then? As long as it doesn't throw and that the following passes, we would know that it's been upgraded successfully:
```js    
it("Should have the correct value for state variable `myVal`", async function () {
      assert.strictEqual(await arrowTokenV2.myVal(), 52, "myVal value");
    });
```

Reply:
To me it looks like we don't need the `beforeEach` because each of the previous `it` statements could mimic the upgrade process instead, which would be easier to reason about.